community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -532,6 +532,6 @@ public static BoltConnector boltConnector( String key )
      */
     private static int dynamicRecordDataSizeForAligningWith( int recordSize )
     {
-        return recordSize - InternalRecordFormatSelector.select().dynamic().getRecordHeaderSize();
+        return recordSize - DynamicRecordFormat.RECORD_HEADER_SIZE;
```

Comment:
Yeah, this thing is a bit odd. The basic problem is that the user config value will include the header size, e.g. represents the whole record size. This will have the data size be different for different formats. We want to be power-of-two aligned and so this status calculation we just shouldn't have here. Dunno what to do with it though.